### PR TITLE
remove unneeded autoanalyze static variable

### PR DESCRIPTION
**Pull request by adizaimi**
* remove unneeded autoanalyze static variable which was added in #801 
* find core regex by stripping dirname from pattern

### CODE REVIEW

File: tests/runtestcase [hunk 1]
```diff
@@ -57,14 +57,16 @@ find_cores() {
     # for all machines in cluster 
     local COREAGE=60
 
-    local cpat=$(< /proc/sys/kernel/core_pattern)  # like cat core_pattern
-    local COREDIR=`dirname $cpat`; 
+    local CPAT=$(< /proc/sys/kernel/core_pattern)  # like cat core_pattern
```
Comment (riverszhang89) [inline tests/runtestcase]: Could you please remove irrelevant changes?

Comment (adizaimi) [reply tests/runtestcase]: They are irrelevant to the cleanup, but relevant otherwise.

Comment (riverszhang89) [reply tests/runtestcase]: Sorry not following. Why is it relevant?

File: tests/runtestcase [hunk 2]
```diff
@@ -57,14 +57,16 @@ find_cores() {
...
+    local COREDIR=`dirname $CPAT`; 
     if [ "x$COREDIR" == "x" ] || [ "x$COREDIR" == "x." ] ; then 
         COREDIR=$TESTDIR
+    else
+        CPAT=`basename $CPAT`
```
[the remark this answers is not in the document]
Comment (riverszhang89) [reply tests/runtestcase]: I really don't want to review irrelevant changes.

**Comment by adizaimi** [comment]
Thanks Rivers.

**Pull request merged**